packages/m2m-gateway/src/api/documentApiConverter.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  agreementApi,
+  catalogApi,
+  eserviceTemplateApi,
+  m2mGatewayApi,
+} from "pagopa-interop-api-clients";
+
+type Document =
+  | catalogApi.EServiceDoc
+  | eserviceTemplateApi.EServiceDoc
+  | agreementApi.Document;
+
+function isAgreementDocument(doc: Document): doc is agreementApi.Document {
+  return "createdAt" in doc;
+}
```

Comment:
I'm not a fan of this.
This is could lead to difficult to  bugs.

The `is` type guard trusts you and does not make any strict typecheck on what your logic does.
If one day we decide to change `agreementApi.Document`'s `createdAt` property typescript would not catch that and we would have a bug difficult to spot.

My proposal is to leave of it was, every domain can have its own `toM2MGatewayApiDocument` and may be fine